Add database notification functionality and configuration variables for schema reloading (#408)
PostgREST/postgrest-docs: configuration.rst

db-pool                  Int     10
db-pool-timeout          Int     10
db-extra-search-path     String  public
db-channel               String  pgrst

> db-channel-enabled       Boolean False

Reviewer: @laurenceisla Now that I remember, we agreed somewhere that `db-channel-enabled` must be True by default. We wanted to be conservative because of pgbouncer, but we already have `db-prepared-statements` as True, which wouldn't work with pgbouncer anyway.

You could make this change in postgrest itself - change `db-channel-enabled` to True by default.

Author: Found the comment where you agreed to it  https://github.com/PostgREST/postgrest/issues/1622#issuecomment-733066644 and Wolfgang gives a very good explanation on to why it should be True by default. I'll make a PR in core and come back to the docs when the changes are merged.

  Multiple schemas can be added in a comma-separated string, e.g. ``public, extensions``.

.. _db-channel:

db-channel
----------

>   The name of the notification channel that PostgREST uses to reload the schema cache. For more information, see :ref:`schema_reloading`.

Reviewer: Note that the db-channel is also used for reloading the config. Check https://github.com/PostgREST/postgrest-docs/issues/409.

.. _db-channel-enabled:

db-channel-enabled
------------------

>   When this is set to :code:`true`, the notification channel specified in :ref:`db-channel` is enabled.

Reviewer: We should mention here that this setting is not compatible with connection poolers using the transaction mode.

  Multiple schemas can be added in a comma-separated string, e.g. ``public, extensions``.

.. _db-channel:

db-channel
----------

>   The name of the notification channel that PostgREST uses to re read the config file and reload the schema cache. For more information, see :ref:`schema_reloading`.

Reviewer: ```suggestion
  The name of the notification channel that PostgREST uses for :ref:`schema_reloading` and configuration reloading.
```

.. _db-channel-enabled:

db-channel-enabled
------------------

  When this is set to :code:`true`, the notification channel specified in :ref:`db-channel` is enabled.

>   Note that this setting is not compatible with connection poolers like PgBouncer in transaction pooling mode.

Reviewer: I think we should remove this for now, since we're not really compatible with pgbouncer.
```suggestion
```
